apps/portals/e2e/explore.spec.ts

```diff
@@ -109,6 +114,109 @@ const getAvailableFacetsDiv = (page: Page) => {
     .filter({ hasText: 'Available Facets' })
 }
 
+const countVisibleElements = async (allCells: Locator, count: number) => {
+  let visibleCount = 0
+  for (let i = 0; i < count; i++) {
+    const cell = allCells.nth(i)
+    if (await cell.isVisible()) {
+      visibleCount++
+    }
+  }
+  return visibleCount
+}
```

Comment:
Seems like we don't need the 2nd parameter, right? See below comments first, we may not need this function

```suggestion
const countVisibleElements = async (allCells: Locator) => {
  const count = await allCells.count()
  let visibleCount = 0
  for (let i = 0; i < count; i++) {
    const cell = allCells.nth(i)
    if (await cell.isVisible()) {
      visibleCount++
    }
  }
  return visibleCount
}
```